cmd/kubeadm/app/util/etcd/etcd.go

```diff
@@ -75,10 +78,49 @@ func New(endpoints []string, ca, cert, key string) (*Client, error) {
 	return &client, nil
 }
 
+// IsEtcdLinsteningOnLocalHostOnly return true if the etcd manifest have etcd listening on localhost only.
+// Listening on local host only was the default in kubeadm <= v1.12, while starting from v1.13 etcd is listening
+// on localhost and API server advertise address (thus allowing add new member when doing join --control-plane).
+func IsEtcdLinsteningOnLocalHostOnly() bool {
```

Comment:
[1]

should we add a TODO that this code is a temporary measure?

Reply:
I would suggest N.B. as a TODO implies an action, but this is only in 1.14 (not in master) and we should not have that problem on 1.14 to 1.15 upgrades.